File: backend/app/models/signal.py
from pydantic import BaseModel

from app.models.market import Direction


class AiSignal(BaseModel):
    id: str
    symbol: str
    direction: Direction
    confidence: int
    entry: float
    stop_loss: float
    take_profit_1: float
    take_profit_2: float
    take_profit_3: float
    risk_reward: float
    reasons: list[str]
    created_at: str
    timeframe: str


class AiAnalysis(BaseModel):
    symbol: str
    ai_score: int
    direction: Direction
    confidence: int
    reasons: list[str]
    entry: float
    stop_loss: float
    take_profit_1: float
    take_profit_2: float
    take_profit_3: float
    risk: float
    reward: float

File: types/signal.ts
import type { Direction } from "./market";

export interface AiSignal {
  id: string;
  symbol: string;
  direction: Direction;
  confidence: number; // 0-100
  entry: number;
  stopLoss: number;
  takeProfit1: number;
  takeProfit2: number;
  takeProfit3: number;
  riskReward: number;
  reasons: string[];
  createdAt: string;
  timeframe: string;
}

